ckan/tests/helpers.py

```diff
@@ -316,21 +371,41 @@ def test_ckan_site_title(self):
 
     :param value: the new config key's value, e.g. ``'My Test CKAN'``
     :type value: string
+
+    .. seealso:: The context manager :py:func:`changed_config`
     '''
     def decorator(func):
+        @functools.wraps(func)
         def wrapper(*args, **kwargs):
-            _original_config = config.copy()
-            config[key] = value
+            with changed_config(key, value):
+                return func(*args, **kwargs)
+        return wrapper
+    return decorator
 
-            try:
-                return_value = func(*args, **kwargs)
-            finally:
-                config.clear()
-                config.update(_original_config)
 
-            return return_value
-        return nose.tools.make_decorator(func)(wrapper)
-    return decorator
+@contextlib.contextmanager
+def changed_config(key, value):
```

Comment:
It does indeed :wink:

`changed_config` is a context-manager-variant of `change_config` (which is a decorator). I added it because I needed to temporarily change the config [in the middle of a test case](https://github.com/torfsen/ckan/blob/2977-rq-background-tasks/ckan/tests/lib/test_jobs.py#L76). That's cumbersome to do with a decorator, so I added a context manager. Note that both variants mention each other in their docstrings and that `change_config` (the decorator) now uses `changed_config` (the context manager) internally, so there's no duplicate code.

I think it's a useful tool, but if you prefer I can drop it (at the cost of making that one test case a bit uglier). In hindsight, the better approach would be to add context manager-capability to the existing `change_config` -- starting from Python 3.2 one can use [`contextlib.ContextDecorator`](https://docs.python.org/3/library/contextlib.html#contextlib.ContextDecorator) for that, on Python 2 we'd have to do it manually (or backport `ContextDecorator`).

So our options are (a) to keep it, (b) to drop it, or (c) to upgrade `change_config`.